saleor/graphql/discount/mutations.py

```diff
@@ -270,7 +270,12 @@ class SaleUpdateMinimalVariantPriceMixin:
     def success_response(cls, instance):
         # Update the "minimal_variant_prices" of the associated, discounted
         # products (including collections and categories).
-        update_products_minimal_variant_prices_of_discount_task.delay(instance.pk)
+        if (
+            instance.products.all().exists()
```

Comment:
`instance.products.exists()`, you don't need `all()`.

Reply:
you right, my bad